src/shogun/features/SparseFeatures.cpp

```diff
@@ -457,13 +457,15 @@ template<> float64_t CSparseFeatures<complex128_t>::dot(int32_t vec_idx1,
 	return 0.0;
 }
 
-template<class ST> float64_t CSparseFeatures<ST>::dense_dot(int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
+template <class ST>
+float64_t
+CSparseFeatures<ST>::dot(int32_t vec_idx1, const SGVector<float64_t> vec2) const
 {
-	REQUIRE(vec2, "dense_dot(vec_idx1=%d,vec2_len=%d): vec2 must not be NULL\n",
-		vec_idx1, vec2_len);
-	REQUIRE(vec2_len>=get_num_features(),
-		"dense_dot(vec_idx1=%d,vec2_len=%d): vec2_len should contain number of features %d %d\n",
-		vec_idx1, vec2_len, get_num_features());
+	REQUIRE(
```

Comment:
basically have this instead (this isn't related to the PR but would be good to clean this up)

Reply:
done

Reply:
cool! thanks!